### PR TITLE
VZ 6271, VZ 6475: Prometheus HA

### PR DESCRIPTION
**Pull request by anders-swanson**
Workaround for VZ-6475 included: Create PVC pre-upgrade and set it to bind with the retained PV for prometheus data migration. Volume claim template selectors are removed, since they block dynamic volume creation.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/common/ingress.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ package common
 import (
 	"context"
 	"fmt"
+	"github.com/verrazzano/verrazzano/pkg/vzmap"
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/common/ingress.go:9]: Import should be moved down, not with standard package imports.

File: platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ package operator
 import (
 	"context"
 	"fmt"
+	"k8s.io/apimachinery/pkg/api/resource"
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go:9]: Move down.

File: platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go [hunk 3]
```diff
@@ -162,12 +177,37 @@ func removeOldClaimFromPrometheusVolume(ctx spi.ComponentContext) error {
...
+			if err := createPVCFromPV(ctx, pv); err != nil {
+				return ctx.Log().ErrorfNewErr("Failed to create new PVC from volume %s: %v", pv.Name, err)
+			}
 			break
 		}
 	}
 	return nil
 }
 
+func createPVCFromPV(ctx spi.ComponentContext, volume corev1.PersistentVolume) error {
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go:189]: Add a function comment?

File: tests/e2e/verify-install/promstack/promstack_test.go [hunk 4]
```diff
@@ -5,6 +5,7 @@ package promstack
 
 import (
 	"fmt"
+	corev1 "k8s.io/api/core/v1"
```
Comment (fredtibbitts) [inline tests/e2e/verify-install/promstack/promstack_test.go:8]: Move down.

**Review by stevshap — APPROVED** [review]

**Pull request merged**